src/lib/model/simple-vo.ts

```diff
@@ -1,4 +1,26 @@
-export interface SimpleVO {
+import { BaseVO } from './base-vo';
+
+export interface SimpleVO extends BaseVO {
   key: string;
-  value: string | boolean;
+  value: string | boolean | PresignedUrlResponse;
 }
+
+export type PresignedUrlResponse = {
+  destinationUrl: string;
+  presignedPost: PresignedPost;
+};
+
+export type PresignedPost = {
+  url: string;
+  fields: PresignedPostFields;
+};
+
+export type PresignedPostFields = {
+  Key: string;
+  bucket: string;
+  'X-Amz-Algorithm': string;
+  'X-Amz-Credential': string;
+  'X-Amz-Date': string;
+  Policy: string;
+  'X-Amz-Signature': string;
+};
```

Comment:
> I tried to do this but I'm not sure if I used the right approach or not - let me know what you think!

This is great! You used exactly the right approach.

Now that I look at it written out, particularly all the `X-` headers, the only thing I'd be cautious of is that we're being too specific about the contents of the `fields` value. That comes from upload-service via the AWS SDK, and so is effectively a black box; it'd probably be safer for us to just call them a set of string/string key-value pairs:

```suggestion
export type PresignedPost = {
  url: string;
  fields: {
    [key: string]: string;
  };
};
```